app/src/main/java/com/google/droidjump/MainActivity.java

```diff
@@ -25,26 +30,59 @@
 public class MainActivity extends AppCompatActivity {
 
     private int levelsCount;
-    private int currentLevel;
+    private SharedPreferences gameData;
+    private SharedPreferences.Editor gameDataEditor;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        gameData = getSharedPreferences(GAME_VIEW_DATA, Context.MODE_PRIVATE);
+        gameDataEditor = gameData.edit();
         levelsCount = GameConstants.GAME_LEVELS_COUNT;
-        currentLevel = 1;
         setContentView(R.layout.main_activity);
-
     }
 
     public int getLevelsCount() {
         return levelsCount;
     }
 
-    public int getCurrentLevel() {
-        return currentLevel;
+    public void increaseLevel() {
+        int currentLevel = getCurrentLevel();
+        int lastLevel = getLastLevel();
+        if (currentLevel < levelsCount) {
+            // Increasing the current level
+            gameDataEditor.putInt(GAME_VIEW_CURRENT_LEVEL_STRING, ++currentLevel);
+            // Increasing the last level
+            if (currentLevel > lastLevel) {
+                gameDataEditor.putInt(GAME_VIEW_LAST_LEVEL_STRING, currentLevel);
+            }
+            gameDataEditor.apply();
+        }
     }
 
     public void setCurrentLevel(int currentLevel) {
-        this.currentLevel = currentLevel;
+        gameDataEditor.putInt(GAME_VIEW_CURRENT_LEVEL_STRING, currentLevel);
+        gameDataEditor.apply();
+    }
+
+    public void setLastLevel(int lastLevel) {
+        gameDataEditor.putInt(GAME_VIEW_LAST_LEVEL_STRING, lastLevel);
+        gameDataEditor.apply();
+    }
+
+    public void resetGameData() {
+        int firstLevel = 1;
+        SharedPreferences.Editor editor = gameData.edit();
+        editor.putInt(GAME_VIEW_CURRENT_LEVEL_STRING, firstLevel);
+        editor.putInt(GAME_VIEW_LAST_LEVEL_STRING, firstLevel);
+        editor.apply();
+    }
+
+    public int getCurrentLevel() {
+        return gameData.getInt(GAME_VIEW_CURRENT_LEVEL_STRING, /* defValue= */ 1);
+    }
+
+    public int getLastLevel() {
+        return gameData.getInt(GAME_VIEW_LAST_LEVEL_STRING, /* defValue= */ 1);
```

Comment:
/* defValue= */ 1 -- extract into a constant. If this is used in multiple files, you can add in the Game Constants file

Reply:
Makes sense. Fixed.